server/runtimewire/runtime_registration.go

```
@@ -12,40 +13,123 @@ type RuntimeRegistry interface {
+
 type BackgroundRouter interface {
 	SetActiveSession(sessionID string, engine *runtime.Engine)
-	ClearActiveSession(sessionID string)
+	ClearActiveSession(sessionID string, engine *runtime.Engine)
 }
 
 type RuntimeRegistration struct {
 	once    sync.Once
 	cleanup func()
+	drain   func(context.Context, func(context.Context) error) error
+	drainMu sync.Mutex
```

golangci-lint (2.12.2)

[error] field drainMu is unused

(unused)